test/unit/spec/PrimaryAuth_spec.js

```diff
@@ -114,10 +115,12 @@ function (Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthForm, Be
     return setup(settings, requests, true);
   }
 
-  function setupPasswordlessAuth (requests) {
-    return setup({ 'features.passwordlessAuth': true }, requests)
+  function setupPasswordlessAuth (requests, refreshState) {
+    return setup({ 'features.passwordlessAuth': true }, requests, refreshState)
       .then(function (test){
-        Util.mockRouterNavigate(test.router);
+        if (!refreshState) {
+          Util.mockRouterNavigate(test.router);
+        }
         test.setNextResponse(resPasswordlessUnauthenticated);
```

Comment:
if `refreshState` is true, then `setup` will set the next response to `resUnauthenticated`, and then here we set up another next response to `resPasswordlessUnauthenticated`... should this be inside the `if`?

Reply:
I think this is needed (tried moving it inside the if, but got some test failures). `resPasswordlessUnauthenticated` is the response form next step of `passwordlessAuth`.

Reply:
ah yeah... so `refreshState` is for when we have a stateToken, so we refresh the state before anything else in the flow. Then this `resPasswordlessUnauthenticated` is for when you actually submit the form with username/identifier only. All good 👍